src/mimeparser.rs

```diff
@@ -1398,6 +1398,17 @@ impl MimeMessage {
             } else {
                 Viewtype::File
             }
+        } else if msg_type == Viewtype::Image || msg_type == Viewtype::Gif {
+            match get_filemeta(decoded_data) {
+                // image is not too big, display as image/gif:
+                Ok((width, height)) if width * height <= constants::MAX_IMAGE_PIXELS => {
+                    part.param.set_int(Param::Width, width as i32);
+                    part.param.set_int(Param::Height, height as i32);
+                    msg_type
+                }
+                // image is too big or size is unknown, display as file:
+                _ => Viewtype::File,
+            }
```

Comment:
This "else if" block is the main change of the PR.
We check image size and set width and height for valid images, otherwise we change msg_type to File.
Setting of width and height previously was done not only for Viewtype::Image and Viewtype::Gif, but for any mime::IMAGE, which is not the same (at least, SVG is mime::IMAGE, but Viewtype::File)